lost/src/test/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImplTest.java

```diff
@@ -219,7 +219,17 @@ public void setMockMode_shouldNotRegisterDuplicateListeners() throws Exception {
         api.requestLocationUpdates(request, listener);
         api.setMockMode(false);
         api.requestLocationUpdates(request, listener);
-        assertThat(shadowLocationManager.getRequestLocationUpdateListeners()).hasSize(2);
+        assertThat(shadowLocationManager.getRequestLocationUpdateListeners()).hasSize(4);
```

Comment:
True, we should only register an engine if it hasn't be registered before. I've updated the code to check if the engine is existing and if not, register it!